web/src/main/java/org/fao/geonet/monitor/onlineresource/wps/OnlineResourceCheckerWps.java

```diff
@@ -0,0 +1,68 @@
+package org.fao.geonet.monitor.onlineresource.wps;
+
+import org.apache.log4j.Logger;
+import org.fao.geonet.monitor.onlineresource.CheckResult;
+import org.fao.geonet.monitor.onlineresource.CheckResultEnum;
+import org.fao.geonet.monitor.onlineresource.OnlineResourceCheckerDefault;
+import org.fao.geonet.monitor.onlineresource.OnlineResourceCheckerUtils;
+import org.jdom.Element;
+import org.joda.time.LocalDateTime;
+import org.joda.time.Period;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+public class OnlineResourceCheckerWps extends OnlineResourceCheckerDefault {
+
+    private static Logger logger = Logger.getLogger(OnlineResourceCheckerWps.class);
+    private String layer;
+    Future<CheckResult> resultFuture;
+
+    //  TODO:  Make configurable
+    public static final int MINUTES_BETWEEN_CHECKS = 1440;
+
+    LocalDateTime lastRun;
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
```

Comment:
Might be better to use a shared executor pool - this could result in as many threads as there are gogoduck layers.

Reply:
The number of threads shouldn't be a problem at this point.